src/util/variant.h

```diff
@@ -0,0 +1,18 @@
+/*******************************************************************\
+
+Module: typedef for variant class template. To be replaced with
+        std::variant once C++17 support is enabled
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_VARIANT_H
+#define CPROVER_UTIL_VARIANT_H
+
+#include <mpark/variant.hpp>
```

Comment:
I suppose this works with CMake, but shouldn't this be `#include <lib/variant/mpark/variant.hpp>`? If that, however, does _not_ work with `CMake` then we'd need to update `INCLUDES` in any `Makefile` that wants to use this.

Reply:
So only the util includes need to change, since all other users should use the `variantt` wrapper. But happy to do that if that's the preference.

Reply:
Well, but the `variant.h` file uses this include, so if I'm in directory `X` and have a file `x/y.cpp` that includes `variant.h` then the preprocessor still needs to know how to resolve that include.